File: src/哈希表/2395.和相等的数组/0326_2395.和相等的数组.js
// 思路 哈希表

/* 知识点：
const hash = new Set(); JS建立哈希表为
hash.add();     哈希表添加数据用add

与数组的区别：
const nums = new Array();
const list = [];
数组判断是否含有某元素为
array.indexOf(arg) (如果不存在返回-1)  ||  arr.indludes(arg)
 */


var findSubarrays = function(nums) {
    // const num = new Array(); 声明数组
    // const num = []
    const num = new Set();
    for(let i = 0; i < nums.length - 1; i++){
        const sum = nums[i] + nums[i + 1]
        // if(num.includes(sum)){
        if(num.has(sum)){
            return true
        } else {
            // num.push(sum)
            num.add(sum)
        }
    }
    return false
};

File: 0326_2395.和相等的数组.py
#思路： 使用哈希表
#建立一个哈希表，用来存储第i个数于第i+1个数之和，
#如果和存在于哈希表中，即为True
# 否则直至遍历结束，返回False

# 知识点
#注意本题中建立的为哈希表set(),set添加数据为 set.add(x)
#数组添加为 nums.append()
def findSubarrays(nums):
    sums = set()
    n = len(nums)
    for i in range(n - 1):
        res = nums[i] + nums[i + 1]
        if res in sums:
            return True
        sums.add(res)
    return False

#执行函数
nums = [4,2,4]
print(findSubarrays(nums))